pkg/container/resolver.go

```diff
@@ -160,3 +181,37 @@ func (r *blockingResolver) Finish() ([]Spec, error) {
 
 	return specs, nil
 }
+
+func (r *blockingResolver) Resolve(source SourceSpec) (Spec, error) {
+	client, err := r.newClient(source.Source)
+	if err != nil {
+		return Spec{}, err
+	}
+
+	client.SetTLSVerify(source.TLSVerify)
+	client.SetArchitectureChoice(r.Arch)
+	if r.AuthFilePath != "" {
+		client.SetAuthFilePath(r.AuthFilePath)
+	}
+
+	return client.Resolve(context.TODO(), source.Name, source.Local)
```

Comment:
Please add a proper context to this method and ideally to `ResolveAll` as well. That way, a timeout can be set globally for the whole operation. I makes sense here (the client) as well as for the async resolver where another `context.Background()` is used.